Updated printer definitions

QORTEC opened the pull request:
# Full List (With Duplicates)
Closes     pdewacht#213 Brother DCP-7010
Closes     pdewacht#141 Brother DCP-7020
Closes     pdewacht#173 Brother DCP-1610W series
Closes     pdewacht#238 Brother DCP-1610W series
Closes     pdewacht#172 Brother DCP-8065DN
Closes     pdewacht#221 Brother DCP-B7500D series
Closes     pdewacht#66  Brother DCP-L2510D series
Closes     pdewacht#192 Brother DCP-L2510D series
Closes     pdewacht#87  Brother DCP-L2537DW
Closes     pdewacht#102 Brother DCP-L2550DW series
Closes     pdewacht#177 Brother DCP-L2550DW series
Closes     pdewacht#210 Brother DCP-L2560DW series
Closes     pdewacht#176 Brother FAX-2820
Closes     pdewacht#189 Brother FAX-2840
Closes     pdewacht#229 Brother HL-1430 series
Closes     pdewacht#63  Brother HL-2130 series
Closes     pdewacht#131 Brother HL-2130 series
Closes     pdewacht#158 Brother HL-2130 series (aka HL-2132)
Closes     pdewacht#147 Brother HL-2230 series
Closes     pdewacht#121 Brother HL-2240D series
Closes     pdewacht#143 Brother HL-2240D series
Closes     pdewacht#194 Brother HL-2240 series
Closes     pdewacht#185 Brother HL-2260
Closes     pdewacht#201 Brother HL-2260D
Closes     pdewacht#101 Brother HL-2250DN series
Closes     pdewacht#128 Brother HL-2280DW
Closes     pdewacht#217 Brother HL-2280DW
Closes     pdewacht#94  Brother HL-5040 series
Closes     pdewacht#193 Brother HL-5140 series
Closes     pdewacht#236 Brother HL-5140 series
Closes     pdewacht#195 Brother HL-5450DN series
Closes     pdewacht#237 Brother HL-5450DN series
Closes     pdewacht#112 Brother HL-L2305 series
Closes     pdewacht#79  Brother HL-L2310D series
Closes     pdewacht#108 Brother HL-L2310D series
Closes     pdewacht#151 Brother HL-L2310D series
Closes     pdewacht#161 Brother HL-L2310D series
Closes     pdewacht#165 Brother HL-L2310D series
Closes     pdewacht#190 Brother HL-L2335D series
Closes     pdewacht#103 Brother HL-L2350DW series
Closes     pdewacht#119 Brother HL-L2350DW series
Closes     pdewacht#120 Brother HL-L2350DW series
Closes     pdewacht#196 Brother HL-L2350DW series
Closes     pdewacht#224 Brother HL-L2350DW series
Closes     pdewacht#234 Brother HL-L2350DW series
Closes     pdewacht#162 Brother HL-L2370DN series
Closes     pdewacht#90  Brother HL-L2380DW series
Closes     pdewacht#231 Brother HL-L2400D
Closes     pdewacht#207 Brother HL-L2400DWE
Closes     pdewacht#205 Brother HL-L2402D
Closes     pdewacht#219 Brother HL-L2402D
Closes     pdewacht#216 Brother HL-L2460DW
Closes     pdewacht#222 Brother HL-L2460DW
Closes     pdewacht#232 Brother HL-L2460DW
Closes     pdewacht#156 Brother HL-L5000D series
Closes     pdewacht#109 Brother MFC-1810 series
Closes     pdewacht#115 Brother MFC-7320
Closes     pdewacht#77  Brother MFC-7340
Closes     pdewacht#84  Brother MFC-7340
Closes     pdewacht#144 Brother MFC-7440N
Closes     pdewacht#187 Brother MFC-7860DW
Closes     pdewacht#208 Brother MFC-8440
Closes     pdewacht#182 Brother MFC-9160
Closes     pdewacht#50  Brother MFC-8710DW
Closes     pdewacht#129 Brother MFC-8860DN
Closes     pdewacht#209 Brother MFC-L2690DW
Closes     pdewacht#134 Brother MFC-L2700DW series
Closes     pdewacht#114 Brother MFC-L2700DN series
Closes     pdewacht#130 Brother MFC-L2710DN series
Closes     pdewacht#227 Brother MFC-L2740DW series
Closes     pdewacht#57  Brother MFC-L2750DW series
Closes     pdewacht#83  Brother MFC-L3750CDW series
Closes     pdewacht#225 Brother MFC-L5800DW series
Closes     pdewacht#142 Lenovo LJ2650DN
Closes     pdewacht#150 Fuji Xerox DocuPrint P265 dw
Closes Owl-Maintain#16  Brother HL-1210W series
Closes Owl-Maintain#31  Brother HL-2150N support
Closes Owl-Maintain#33  Brother HL-5250DN series
Closes Owl-Maintain#45  Brother HL-5350DN series
Closes Owl-Maintain#8   Brother HL-5370DW series
Closes Owl-Maintain#48  Brother HL-L1232W
Closes Owl-Maintain#47  Brother HL-L2325DW
Closes Owl-Maintain#42  Brother HL-L2400D
Closes Owl-Maintain#20  Brother HL-L2400DW
Closes Owl-Maintain#41  Brother HL-L2400DWE
Closes Owl-Maintain#14  Brother HL-L2405W
Closes Owl-Maintain#49  Brother HL-L2440DW
Closes Owl-Maintain#30  Brother HL-L2460DW
Closes Owl-Maintain#32  Brother HL-L2480DW
Closes Owl-Maintain#50  Brother MFC-L2685DW
Closes Owl-Maintain#36  Brother MFC-L2800DW

---------------------

# Original Post

Added the following printer definitions

Pulled #107 Brother DCP-7070DW Thanks @musinsky

Closes #141 Brother DCP-7020
Closes #172 Brother DCP-8065DN
Closes #66  Brother DCP-L2510D series
Closes #87  Brother DCP-L2537DW
Closes #102 Brother DCP-L2550DW series
Closes #63  Brother HL-2130  series
Closes #147 Brother HL-2230  series
Closes #121 Brother HL-2240D series
Closes #101 Brother HL-2250DN series
Closes #128 Brother HL-2280DW
Closes #94  Brother HL-5040  series
Closes #112 Brother HL-L2305 series
Closes #108 Brother HL-L2310D  series
Closes #103 Brother HL-L2350DW series
Closes #162 Brother HL-L2370DN series
Closes #90  Brother HL-L2380DW series
Closes #156 Brother HL-L5000D series
Closes #109 Brother MFC-1810  series
Closes #115 Brother MFC-7320
Closes #77  Brother MFC-7340
Closes #144 Brother MFC-7440N
Closes #50  Brother MFC-8710DW
Closes #129 Brother MFC-8860DN
Closes #134 Brother MFC-L2700DW series
Closes #114 Brother MFC-L2700DN series
Closes #130 Brother MFC-L2710DN series
Closes #57  Brother MFC-L2750DW series
Closes #142 Lenovo LJ2650DN
Closes #150 Fuji Xerox DocuPrint P265 dw

QORTEC committed:
Add new printer definitions:
Closes pdewacht#141 Brother DCP-7020
Closes pdewacht#66  Brother DCP-L2510D  series
Closes pdewacht#87  Brother DCP-L2537DW
Closes pdewacht#102 Brother DCP-L2550DW series
Closes pdewacht#63  Brother HL-2130 series
Closes pdewacht#147 Brother HL-2230 series
Closes pdewacht#121 Brother HL-2240D  series
Closes pdewacht#101 Brother HL-2250DN series
Closes pdewacht#128 Brother HL-2280DW
Closes pdewacht#94  Brother HL-5040  series
Closes pdewacht#112 Brother HL-L2305 series
Closes pdewacht#108 Brother HL-L2310D  series
Closes pdewacht#103 Brother HL-L2350DW series
Closes pdewacht#90  Brother HL-L2380DW series
Closes pdewacht#109 Brother MFC-1810   series
Closes pdewacht#115 Brother MFC-7320
Closes pdewacht#77  Brother MFC-7340
Closes pdewacht#144 Brother MFC-7440N
Closes pdewacht#50  Brother MFC-8710DW
Closes pdewacht#129 Brother MFC-8860DN
Closes pdewacht#114 Brother MFC-L2700DN series
Closes pdewacht#134 Brother MFC-L2700DW series
Closes pdewacht#130 Brother MFC-L2710DN series
Closes pdewacht#57  Brother MFC-L2750DW series
Closes pdewacht#150 Fuji Xerox DocuPrint P265 dw
Closes pdewacht#142 Lenovo LJ2650DN

musinsky committed:
Add Brother DCP-7070DW

QORTEC committed:
Add new printer definitions:
Closes pdewacht#173 Brother DCP-1610W series
Closes pdewacht#176 Brother FAX-2820

QORTEC committed:
Add new printer definitions:
Closes pdewacht#172 Brother DCP-8065DN
Closes pdewacht#162 Brother HL-L2370DN series
Closes pdewacht#156 Brother HL-L5000D  series

QORTEC committed:
Add new printer definitions:
Closes pdewacht#189 Brother FAX-2840
Closes pdewacht#185 Brother HL-2260
Closes pdewacht#190 Brother HL-L2335D series
Closes pdewacht#187 Brother MFC-7860DW
Closes pdewacht#182 Brother MFC-9160

QORTEC commented:
Added the following printer definitions

Closes #173 Brother DCP-1610W series
Closes #176 Brother FAX-2820
Closes #189 Brother FAX-2840
Closes #185 Brother HL-2260
Closes #190 Brother HL-L2335D series
Closes #187 Brother MFC-7860DW
Closes #182 Brother MFC-9160

qwertychouskie commented:
@pdewacht Any chance of this getting merged soonish and a new release tagged?  This would greatly benefit the owners of a variety of printers (including me).

qwertychouskie commented:
(Also @QORTEC this PR shows merge conflicts for some reason, perhaps a bad rebase?)

QORTEC commented:
@qwertychouskie your correct; when I rebased I apparently accidentally "recommitted" f3de875079e0544e91eb9992bf308a78545b23c5 (which was already in `pdewacht/brlaser`) which seems to be source of the conflict.

I'll fix that later on when I have more time, though "functionally" it shouldn't matter, and that commit can be ignored when pulling the changes from my fork.

While its important to resolve this conflict, I consider it to be "low priority" since @pdewacht doesn't seam to have time to maintain `brlaser` at the moment and a quick resolution doesn't have a functional benefit. Package maintainers will probably ether use vanilla `brlaser` found here or a release on `Owl-Maintain/brlaser` with the patches already applied.

erooke committed:
Add hl-5370dw support

**Printer Entry**:
````
Device: uri = usb://Brother/HL-5370DW%20series?serial=C2J486800
        class = direct
        info = Brother HL-5370DW series
        make-and-model = Brother HL-5370DW series
        device-id = MFG:Brother;CMD:PJL,PCL,PCLXL;MDL:HL-5370DW series;CLS:PRINTER;
        location =
````

**Verified**:
- [x] test page printed
- [x] tested 300dpi
- [x] tested 600dpi
- [x] tested 1200dpi
- [x] texted simplex <!-- single-sided printing -->
- [x] tested duplex  <!-- if applicable -->

**Additional notes**:
Contrary to the other printers the duplex directive had to be set to
normal to get correct duplex printing to work.

hifron commented:
Currently package **printer-driver-brlaser** is installed in Ubuntu but also there is driver from Brother which should be more beneficial to Brother users who wants to print something as it seems as Rust is preferred language for @pdewacht now but also Brother for older printer seems not so interested for Linux(but other systems seems not so)... 

What about transfer this repo to Brother organization or OpenPrinting and let help maintain them with not so overhead for currently Linux drivers with some support from Brother for fixing bugs? And maybe make OpenPrinting App which would somehow work with nowadays/3.0 CUPS and maybe make it more official with Brother. @michaelrsweet ?

But one huge problem(for Brother organization) could be that some printers also have scanners and Scanner application is not installed by default either(but SANE app installable) or possible with current CUPS(but planned) and Brother have builtin combined drivers(but maybe outdated)... 

**But if faulty installed package printer-driver-brlaser is problematic and not maintained, its disaster for some users.**

michaelrsweet commented:
@hifron I can't speak for the OpenPrinting organization, but I know for me that I don't have the time or resources (printers, documentation, etc.) to support such an effort directly. The best I can do is advise developers when they have questions, either for CUPS driver developer or migration to a printer application using PAPPL.

hifron commented:
https://www.turboprint.info/ may be helpful if there is no interest and official drivers seems not suitable :)

QORTEC committed:
Add new printer definitions:
Closes pdewacht#213 Brother DCP-7010
Closes pdewacht#221 Brother DCP-B7500D series
Closes pdewacht#210 Brother DCP-L2560DW series
Closes pdewacht#194 Brother HL-2240 series
Closes pdewacht#201 Brother HL-2260D
Closes pdewacht#193 Brother HL-5140 series
Closes pdewacht#195 Brother HL-5450DN series
Closes pdewacht#205 Brother HL-L2402D
Closes pdewacht#208 Brother MFC-8440
Closes pdewacht#209 Brother MFC-L2690DW
Closes Owl-Maintain#20  Brother HL-L2400DW
Closes Owl-Maintain#14  Brother HL-L2405W

Fix DocuPrint P265 dw PPD file name

QORTEC commented:
Added the following printer definitions: 

Closes pdewacht#213 Brother DCP-7010
Closes pdewacht#221 Brother DCP-B7500D series
Closes pdewacht#210 Brother DCP-L2560DW series
Closes pdewacht#194 Brother HL-2240 series
Closes pdewacht#201 Brother HL-2260D
Closes pdewacht#193 Brother HL-5140 series
Closes pdewacht#195 Brother HL-5450DN series
Closes pdewacht#205 Brother HL-L2402D
Closes pdewacht#208 Brother MFC-8440
Closes pdewacht#209 Brother MFC-L2690DW
Closes Owl-Maintain#8   Brother HL-5370DW series
Closes Owl-Maintain#20  Brother HL-L2400DW
Closes Owl-Maintain#14  Brother HL-L2405W

QORTEC committed:
Switch from `Duplex normal` to `Duplex rotated`

Deepdive2 commented:
Here is the output of my Brother 2310D witch is working. The only thing is I can only choose 600 or 1200dpi. For txt this is ok but for pictures I would also need 300dpi. With 600dpi the printed doen’t print pictures. Would be nice if this you’ll be implemented. Thanks a lot. 

Device: uri = usb://Brother/HL-L2310D%20series?serial=E78096J8N675866
        class = direct
        info = Brother HL-L2310D series
        make-and-model = Brother HL-L2310D series
        device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-L2310D series;CLS:PRINTER;CID:Brother Laser Type1;
        location =

QORTEC commented:
@Deepdive2 
Your printer, Brother HL-L2310D series, is already in this pull request.
To enable 300DPI can you please confirm that it works?
You can use the HL-2270DW driver, it has both 300DIP and Duplex enabled.

onny committed:
Add HL-2150N support

alextwothousand commented:
Reckon it'd be worth making your own fork @QORTEC? There's a good few printers that could be supported...

Edit: Just noticed you have done - sorry!

QORTEC commented:
> There's a good few printers that could be supported...

A large portion of Linux distributions now package the _Owl-Maintain_ fork for their repos instead, you can see list [here](/Owl-Maintain/brlaser/wiki/Installing-brlaser).

> Edit: Just noticed you have done - sorry!

Yeah, the **fork** is not really "_visible_".
I've had to rename the driver to `Owl-Maintain/brlaser`, since there is a large number of people who still come here to open `Issues` even though they're using the fork.

QORTEC committed:
Add new printer definitions:
Closes pdewacht#227 Brother MFC-L2740DW series
Closes pdewacht#225 Brother MFC-L5800DW series
Closes Owl-Maintain#16 Brother HL-1210W series
Closes Owl-Maintain#33 Brother HL-5250DN series
Closes Owl-Maintain#30 Brother HL-L2460DW
Closes Owl-Maintain#32 Brother HL-L2480DW
Closes Owl-Maintain#36 Brother MFC-L2800DW

hifron commented:
there could be repo transfer request... but official support could also take account supportive issue... org, out of the box working, but that is issue itself...

QORTEC committed:
Add new printer definitions:
Closes pdewacht#229 Brother HL-1430 series
Closes Owl-Maintain#45 Brother HL-5350DN series
Closes Owl-Maintain#48 Brother HL-L1232W
Closes Owl-Maintain#47 Brother HL-L2325DW
Closes Owl-Maintain#42 Brother HL-L2400D
Closes Owl-Maintain#41 Brother HL-L2400DWE
Closes Owl-Maintain#49 Brother HL-L2440DW
Closes Owl-Maintain#50 Brother MFC-L2685DW

QORTEC committed:
Update README.md

Updated “Supported Printers” selection:
- Updated the printer list to include the newly added printers.

QORTEC committed:
Standardized printer naming, and list only supported printers

Updated README.md
- `Brother DCP-7080D` was added to the supported printers list
- `Brother MFC-1910W series` was fixed to indicate that it is a 'series'
- `Brother MFC-7840W` & `Lenovo M7605D` were removed because their definitions are not available

Update brlaser.drv.in
- Use standard naming convention for ModelName & NickName
- Corrected ModelName (1284DeviceID > MDL):
  - DCP-1510 series
  - DCP-L2500D series
  - DCP-L2520D series
  - DCP-L2520DW series
  - DCP-L2540DW series
  - HL-1110 series
  - HL-1200 series
  - HL-L2300D series
  - HL-L2320D series
  - HL-L2340D series
  - HL-L2360D series
  - HL-L2375DW series
  - MFC-1910W series
- Corrected NickName (1284DeviceID > MFG MDL):
  - Brother MFC-1910W series
  - Brother MFC-7420

QORTEC commented:
Added the following printer definitions:
Closes pdewacht#229 Brother HL-1430 series
Closes pdewacht#227 Brother MFC-L2740DW series
Closes pdewacht#225 Brother MFC-L5800DW series
Closes Owl-Maintain#16 Brother HL-1210W series
Closes Owl-Maintain#31 Brother HL-2150N support
Closes Owl-Maintain#33 Brother HL-5250DN series
Closes Owl-Maintain#45 Brother HL-5350DN series
Closes Owl-Maintain#48 Brother HL-L1232W
Closes Owl-Maintain#47 Brother HL-L2325DW
Closes Owl-Maintain#42 Brother HL-L2400D
Closes Owl-Maintain#41 Brother HL-L2400DWE
Closes Owl-Maintain#49 Brother HL-L2440DW
Closes Owl-Maintain#30 Brother HL-L2460DW
Closes Owl-Maintain#32 Brother HL-L2480DW
Closes Owl-Maintain#50 Brother MFC-L2685DW
Closes Owl-Maintain#36 Brother MFC-L2800DW

QORTEC committed:
Add missing 300dpi option to select printers
- HL-5250DN
- HL-L2480DW
- MFC-8860DN
- MFC-L2800DW
